From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Thu, 24 Sep 2026 11:23:20 -0400
Subject: [PATCH] cmake: Make iio-compat define LIBIIO1_NAME to exact target
 name

This moves what was already done for the Windows library and uses it for
everything, ensuring that the library that the compatibility shim looks
up is the same name as what was built no matter what customizations are
made.

Signed-off-by: Ryan Volz <ryan.volz@gmail.com>
---
 CMakeLists.txt | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index af08deae..71fef7be 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -140,19 +140,12 @@ if (CMAKE_C_COMPILER_ID MATCHES "GNU|Clang|MSVC")
         add_library(iio-compat compat.c)
         if (WIN32)
             target_sources(iio-compat PRIVATE dynamic-windows.c)
-            set(LIBIIO1_NAME "$<TARGET_FILE_NAME:iio>")
         else ()
             target_sources(iio-compat PRIVATE dynamic-unix.c)
             target_link_libraries(iio-compat PRIVATE dl)
-            if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-                set(LIBIIO1_NAME
-                    "iio.framework/Versions/${PROJECT_VERSION}/iio")
-            else ()
-                set(LIBIIO1_NAME "libiio.so.${PROJECT_VERSION_MAJOR}")
-            endif ()
         endif ()
         target_compile_definitions(iio-compat
-                                   PRIVATE LIBIIO1_NAME="${LIBIIO1_NAME}")
+                                   PRIVATE LIBIIO1_NAME="$<TARGET_FILE_NAME:iio>")
 
         target_include_directories(iio-compat PUBLIC ${CMAKE_BINARY_DIR}
                                                      include)
-- 
2.51.0

